Model/Indexer/Data.php

```diff
@@ -57,35 +67,73 @@ class Data implements IndexerActionInterface, MviewActionInterface
     /** @var NostoHelperAccount */
     private $nostoHelperAccount;
 
+    /** @var NostoHelperScope */
+    private $nostoHelperScope;
+
+    /** @var DimensionProviderInterface; */
+    private $dimensionProvider;
+
+    /** @var ModeSwitcher */
+    private $modeSwitcher;
+
     /** @var NostoLogger */
     private $nostoLogger;
 
+    /** @var ProcessManager */
+    private $processManager;
+
     /**
      * Data constructor.
      * @param NostoIndexService $nostoServiceIndex
      * @param NostoHelperAccount $nostoHelperAccount
+     * @param NostoHelperScope $nostoHelperScope
+     * @param DimensionProviderInterface $dimensionProvider
+     * @param ModeSwitcher $modeSwitcher
      * @param NostoLogger $nostoLogger
+     * @param ProcessManager|null $processManager
      */
     public function __construct(
         NostoIndexService $nostoServiceIndex,
         NostoHelperAccount $nostoHelperAccount,
-        NostoLogger $nostoLogger
+        NostoHelperScope $nostoHelperScope,
+        DimensionProviderInterface $dimensionProvider,
+        ModeSwitcher $modeSwitcher,
+        NostoLogger $nostoLogger,
+        ProcessManager $processManager = null
     ) {
         $this->nostoServiceIndex = $nostoServiceIndex;
         $this->nostoHelperAccount = $nostoHelperAccount;
+        $this->nostoHelperScope = $nostoHelperScope;
+        $this->dimensionProvider = $dimensionProvider;
+        $this->modeSwitcher = $modeSwitcher;
         $this->nostoLogger = $nostoLogger;
+        $this->processManager = $processManager;
     }
 
     /**
      * @inheritdoc
      * @throws Exception
      */
     public function executeFull()
+    {
+        $dimensionProvider = $this->getDimensionsProvider();
```

Comment:
Simply call `$this->executeInParallel($dimensionProvider);` Doesn't matter if `dimensionProvider` is null.

Reply:
Remove these null checks from here.

Reply:
Don't think that a null value can be passed. $dimension should be an array as declared in the `DimensionalIndexerInterface`, 
`executeByDimensions(array $dimensions, \Traversable $entityIds)`